### PR TITLE
Implemented Algolia search

### PR DESCRIPTION
**Pull request by leroy**
Implements Algolia search as discussed in #4461

### CODE REVIEW

File: conf.py [hunk 1]
```diff
@@ -24,8 +24,8 @@
 if not os.getenv("SPHINX_NO_GDSCRIPT"):
     extensions.append("gdscript")
 
-if not os.getenv("SPHINX_NO_SEARCH"):
-    extensions.append("sphinx_search.extension")
+# if not os.getenv("SPHINX_NO_SEARCH"):
```
Comment (Calinou) [inline conf.py:27]: Please remove this code entirely instead of commenting it out.

File: conf.py [hunk 2]
```diff
@@ -24,8 +24,8 @@
...
+    # extensions.append("sphinx_search.extension")
```
Comment (Calinou) [inline conf.py:28]: Same as above.

**Review by Calinou — CHANGES_REQUESTED** [review]

File: _static/css/algolia.css [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.wy-nav-side { overflow: visible; }
+.wy-side-scroll { overflow: inherit; }
+
+.algolia-autocomplete {
+    display: block !important;
+}
```
Comment (Calinou) [inline _static/css/algolia.css:6]:
```suggestion
}

```

**Comment by leroy** [comment]
@Calinou

Hey :)

Do you need me to do anything to get this PR merged?

**Comment by Calinou** [comment]
@leroy There are merge conflicts to solve in `requirements.txt`. Apologies for the delay.

**Comment by leroy** [comment]
Hey no worries at all. I've resolved the merge conflicts.

**Review by Calinou — APPROVED** [review]
Thanks!

**Comment by Calinou** [comment]
It seems to work on the [`latest` branch](docs.godotengine.org/en/latest/):

image

There are some things we need to improve still:

- Pressing the <kbd>Enter</kbd> key while the field is empty shouldn't do anything, since it will redirect you to a standard search page with no results found.
- `ref` badges should be placed on the same line as the page name (at the left of the page name), rather than on a separate line.
- There should be a dark theme when the documentation uses the dark theme as well (defined via CSS' `prefers-color-scheme: dark` media query).

**Pull request merged**